server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1229,31 +1234,33 @@ public static List<AliasMetadata> resolveAndValidateAliases(
      * Creates the index into the cluster state applying the provided blocks. The final cluster state will contain an updated routing
      * table based on the live nodes.
      */
-    static ClusterState clusterStateCreateIndex(
-        ClusterState currentState,
+    static void clusterStateCreateIndex(
+        Metadata currentMetadata,
+        ClusterBlocks currentClusterBlocks,
+        RoutingTable currentRoutingTable,
```

Comment:
Rather than pass in the current cluster state, just pass in the required in-parameters.